sycl/include/sycl/detail/defines.hpp

```diff
@@ -27,7 +27,8 @@
 #define __SYCL_SPECIAL_CLASS
 #endif
 
-#if __has_cpp_attribute(__sycl_detail__::sycl_type)
+#if __has_cpp_attribute(__sycl_detail__::sycl_type) &&                         \
```

Comment:
Can you add a FIXME here and above? Something like - ` Check for  __SYCL_DEVICE_ONLY__ can be removed if implementation of __has_attribute is fixed to consider LangOpts when generating attributes in tablegen.`

Reply:
Done.